.github/workflows/main.yml

```diff
@@ -1,50 +1,120 @@
+# Can be run locally using act and docker, or in Fedora 39 also with podman, using:
+# act --bind --container-daemon-socket $XDG_RUNTIME_DIR/podman/podman.sock -W .github/workflows/main.yml
+
 name: Unit tests
 
-on: [push, pull_request]
+on: pull_request
+env:
+  PYTHONWARNINGS: "ignore"
 
 jobs:
-  test_py2:
-    runs-on: ubuntu-20.04
-
+  test:
+    strategy:
+      max-parallel: 1
+      matrix:
+        include:
+        - python-version: '3.11'
+          os: ubuntu-latest
+        - python-version: '2.7'
+          os: ubuntu-20.04
+        - python-version: '3.10'
+          os: ubuntu-latest
+        - python-version: '3.9'
+          os: ubuntu-latest
+        - python-version: '3.8'
+          os: ubuntu-latest
+        - python-version: '3.7'
+          os: ubuntu-latest
+        - python-version: '3.6'
+          os: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
+    permissions:
+      # Gives the action the necessary permissions for publishing new
+      # comments in pull requests.
+      pull-requests: write
+      # Gives the action the necessary permissions for pushing data to the
+      # python-coverage-comment-action branch, and for editing existing
+      # comments (to avoid publishing multiple comments in the same PR)
+      contents: write
```

Comment:
Correct, good catch, was a leftover from another action. I'll remove it!

Reply:
I missed the pylint reported, maybe the permissions are needed for that?

Reply:
They could have been needed for actions adding/updating their coverage comments to/in the PR (and making commits):

Specfically, it was from the examples for this action:
https://github.com/py-cov-action/python-coverage-comment-action

It could have been nice because it would commit the coverate reports into a branch of the repo...
... and the htmlpreview of the generated html could be put into the README.md of the repo like in this example:

https://htmlpreview.github.io/?https://github.com/py-cov-action/python-coverage-comment-action-v3-example/blob/python-coverage-comment-action-data/htmlcov/index.html -

But, this action failed to find the .coverage / coverage.xml files, so I could not get it to work yet,
and removed it from the workflow.

Again, with codecov and possibly other possibilities obsolete, and I'll remove it at least for now!